src/units/unit.cpp

```diff
@@ -2116,9 +2269,12 @@ void unit::apply_builtin_effect(std::string apply_to, const config& effect)
 	} else if(apply_to == "remove_ability") {
 		if(const config& ab_effect = effect.child("abilities")) {
 			for(const config::any_child ab : ab_effect.all_children_range()) {
-				remove_ability_by_id(ab.cfg["id"]);
+				remove_ability_by_attribute(ab.cfg, false, ab.key);
```

Comment:
Ultimately, if we're adding a `[filter_ability]` tag that is intended to replace the old method, I think the best approach is to keep this line unchanged and retain the old `remove_ability_by_id` function, and add a deprecation message if the `[abilities]` tag exists. If I'm understanding the code correctly, this would allow you to remove the `split_filter_as_list` parameter, right? That would also make the code easier to read.

I also have one other concern about the code readability, but I'll hold off on that for now. I'll mention it later.

Reply:
it's done except for deprecation message

Reply:
message added